src/pages/workspaces/workspace/ShareWorkspaceModal/Collaborator.test.ts

```diff
@@ -0,0 +1,225 @@
+  });
+
+  it('shows the role selection as disabled for the current user ', () => {
+    const item: AccessEntry = {
+      email: 'owner@test.com',
+      pending: false,
+      canShare: true,
+      canCompute: true,
+      accessLevel: 'OWNER',
+    };
+    const setAcl: Dispatch<SetStateAction<WorkspaceAcl>> = asMockedFn(jest.fn());
+    const acl = [item];
+
+    render(
+      h(Collaborator, {
+        aclItem: item,
+        acl,
+        setAcl,
+        originalAcl: acl,
+        workspace,
+        lastAddedEmail: undefined,
+      })
+    );
+
+    const select = screen.getByLabelText(`permissions for ${item.email}`);
+    expect(select).not.toBeNull();
+    expect(select).toBeDisabled();
+  });
+
+  it('shows a selection of available access levels', async () => {
+    const item: AccessEntry = {
+      email: 'abc@test.com',
+      pending: false,
+      canShare: true,
+      canCompute: true,
+      accessLevel: 'OWNER',
+    };
+    const setAcl: Dispatch<SetStateAction<WorkspaceAcl>> = asMockedFn(jest.fn());
+    const acl = [item];
+
+    render(
+      h(Collaborator, {
+        aclItem: item,
+        acl,
+        setAcl,
+        originalAcl: acl,
+        workspace: { ...workspace, accessLevel: 'PROJECT_OWNER' },
+      })
+    );
+    const select = screen.getByLabelText(`permissions for ${item.email}`);
+    expect(select).not.toBeNull();
+    expect(select).not.toBeDisabled();
+
+    // The user's current permission is displayed by default in the select component
+    // Opening the menu causes it to be rendered twice in the DOM:
+    // once for the default display, and once for the menu option
+    expect(screen.queryByText('Owner')).not.toBeNull();
+    await openMenu(item.email);
+    expect(screen.queryByText('Writer')).not.toBeNull();
+    expect(screen.queryByText('Reader')).not.toBeNull();
+    expect(screen.queryByText('Project Owner')).toBeNull();
+  });
+
+  it('removes a user with setAcl with the x is selected', async () => {
+    const removeItem: AccessEntry = {
+      email: 'user1@test.com',
+      pending: false,
+      canShare: true,
+      canCompute: true,
+      accessLevel: 'OWNER',
+    };
+    const item: AccessEntry = {
+      email: 'user2@test.com',
+      pending: false,
+      canShare: true,
+      canCompute: true,
+      accessLevel: 'OWNER',
+    };
+
+    const setAcl = jest.fn((val) => {
+      expect(val).not.toContainEqual(removeItem);
+    });
+    const aclMock = asMockedFn(setAcl);
+    const acl = [item, removeItem];
+
+    render(
+      h(Collaborator, {
+        aclItem: removeItem,
+        acl,
+        setAcl: aclMock,
+        originalAcl: acl,
+        workspace: { ...workspace, accessLevel: 'PROJECT_OWNER' },
+      })
+    );
+
+    expect(screen.getByLabelText(`permissions for ${removeItem.email}`)).not.toBeNull();
+    const removeButton = screen.getByRole('button'); // this appears to be the only button in the component
```

Comment:
It would be nice if the aria-label actually included the name of the user who would be removed. When there are multiple users in the dialog, you will get multiple buttons that are all labeled "Remove".

<img width="459" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/484484/3b334079-deca-4b0e-9fc1-c1a63eafeba1">

Reply:
It looks like there are other a11y issues with this dialog too. Probably best for us to create a specific JIRA bug for these problems and leave this PR as a more straightforward conversion of the existing JS.

<img width="830" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/484484/eed0ed04-694f-4e58-bb32-b2fdc5cac9f0">

Reply:
Created https://broadworkbench.atlassian.net/browse/WOR-1112